post_office/backends.py

```diff
@@ -28,14 +28,20 @@ def send_messages(self, email_messages):
         for email_message in email_messages:
             subject = email_message.subject
             from_email = email_message.from_email
-            message = email_message.body
             headers = email_message.extra_headers
 
-            # Check whether email has 'text/html' alternative
-            alternatives = getattr(email_message, 'alternatives', ())
-            for alternative in alternatives:
-                if alternative[1].startswith('text/html'):
-                    html_message = alternative[0]
+            # Look for first 'text/plain' alternative in email
+            for part in email_message.message().walk():
```

Comment:
Do you mind caching `email_message.message()` to a separate variable so `message()` is not called twice? This method [does quite a few things](https://github.com/django/django/blob/235b68135172c473e022045931719f047cf3dc4a/django/core/mail/message.py#L236) so it'd be more efficient if we cache it.

It would be even better if we can avoid walking over `message()` twice. Something like this perhaps?
```python
message = ''
html_message = ''
for part in email_message.message().walk():
    if part.get_content_type() == 'text/plain':
        message = part.get_payload()
    if part.get_content_type() == 'text/html':
        html_message = part.get_payload()
```

Reply:
First, I attempted exactly this approach. The problem is that we then don't know if a message is plaintext, HTML or both. In addition we want to use the first parts, not the last ones. Doing so with only one loop is more complicate.

Moreover, since both loops usually break after one or two iterations, having two loop probably is faster than not breaking and iterating over all attachments.

Reply:
fixed